File: BOJ/10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main()
{
	string s;
	stack<int> sk;
	int n;
	cin >> n;

	while (n--)
	{
		cin >> s;
		if (s == "push")
		{
			int tmp;
			cin >> tmp;
			sk.push(tmp);
		}
		else if(s == "top")
		{
			if (sk.empty() == 0)
			{
				cout << sk.top() << endl;
			}
			else
			{
				cout << -1 << endl;
			}
		}
		else if (s == "size")
		{
			cout << sk.size() << endl;
		}
		else if (s == "empty")
		{
			cout << sk.empty() << endl;
		}
		else if (s == "pop")
		{
			if (sk.empty() ==0)
			{
				cout << sk.top() << endl;
				sk.pop();
			}
			else
			{
				cout << -1 << endl;
			}
		}
	}
	return 0;
}

File: BOJ/10828.py
import sys

n = int(sys.stdin.readline())
st = []

for i in range(n):
    inst = sys.stdin.readline().split()
    if inst[0] == 'push':
        st.append(int(inst[1]))
    elif inst[0] == 'pop':
        if len(st) == 0:
            print(-1)
        else:
            print(st[-1])
            st.pop()
    elif inst[0] == 'size':
        print(len(st))
    elif inst[0] == 'empty':
        if len(st) == 0:
            print(1)
        else:
            print(0)
    elif inst[0] == 'top':
        if len(st) == 0:
            print(-1)
        else:
            print(st[-1])